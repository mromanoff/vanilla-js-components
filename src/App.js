import Component from "./Component";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { ...props }; // Spread in the props from our initial state as the state.
  }

  template(props) {
    // the data is passed in as props here.
    return `
        <div class="App">
          <header class="App-header" data-component="header"></header>
          <main class="App-main" data-component="main"></main>
          <footer class="App-footer" data-component="footer"></footer>
        </div>
      `;
  }
}

export default App;
