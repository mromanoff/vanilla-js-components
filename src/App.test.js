/**
 * @jest-environment jsdom
 */

import App from './App';

jest.mock('./Stopwatch');


it('App renders correctly', () => {

  document.componentRegistry = {};
  document.nextId = 0;

  const element = document.createElement('div');
  element.setAttribute('id', 'root');
  let component = new App({element});
  component.render();

  expect(component).toMatchSnapshot();
});