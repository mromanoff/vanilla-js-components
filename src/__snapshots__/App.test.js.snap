// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
App {
  "_id": 1,
  "state": Object {
    "element": <div
      id="root"
    >
      
        
      <div
        class="App"
      >
        
          
        <header
          class="App-header"
          data-component="header"
        />
        
          
        <main
          class="App-main"
          data-component="main"
        />
        
          
        <footer
          class="App-footer"
          data-component="footer"
        />
        
        
      </div>
      
      
    </div>,
  },
}
`;
